dashboard/app/models/pd/attendance.rb

```diff
@@ -19,7 +19,19 @@ class Pd::Attendance < ActiveRecord::Base
 
   has_one :workshop, class_name: 'Pd::Workshop', through: :session
 
-  def self.for_teacher_in_workshop(teacher, workshop)
-    joins(:workshop).where(teacher: teacher, pd_workshops: {id: workshop.id})
+  def self.for_teacher(teacher)
+    self.joins(:workshop).where(teacher_id: teacher.id)
+  end
```

Comment:
Nit: prefer [scopes](http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope) over class methods here.  (Scopes are just syntactic sugar for the class method.)